File: webapp/codigos-fonte/Java/Vetor.java
package Java;

public class Vetor {
    private final int TAMANHO = 10;
    private final Integer[] vet = new Integer[TAMANHO];

    public int getTAMANHO() {
        return TAMANHO;
    }

    public Integer[] getVet() {
        return vet;
    }

    public void inserir(int valor, int indice) {
        if(indice < TAMANHO) {
            vet[indice] = valor;
        }
    }

    public void printAll() {
        for(int i = 0; i < TAMANHO; i++) {
            System.out.printf("%d ", vet[i]);
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Vetor v = new Vetor();
        v.inserir(15, 1);
        v.printAll();
        v.inserir(16, 0);
        v.printAll();
        v.inserir(151, 1);
        v.printAll();
        v.inserir(152, 0);
        v.printAll();
        v.inserir(153, 1);
        v.printAll();
        v.inserir(154, 1);
        v.printAll();
        v.inserir(155, 1);
        v.printAll();
    }
}

File: webapp/codigos-fonte/Kotlin/vetor.kt
package vetor

data class Vetor (
        val TAMANHO:Int = 10,
        val vetor:Array<Int?> = arrayOfNulls(TAMANHO)
) {
    fun inserir(valor:Int, indice:Int) {
        if(indice < TAMANHO) {
            vetor[indice] = valor
        }
    }

    fun printAll() {
        for (i in 0 until TAMANHO) {
            print(vetor[i].toString() + " ")
        }
        println()
    }
}

fun main() {
    val v = Vetor()
    v.inserir(15, 1)
    v.printAll()
    v.inserir(16, 0)
    v.printAll()
    v.inserir(151, 1)
    v.printAll()
    v.inserir(152, 0)
    v.printAll()
    v.inserir(153, 1)
    v.printAll()
    v.inserir(154, 1)
    v.printAll()
    v.inserir(155, 1)
    v.printAll()
}